meson.build

```diff
@@ -0,0 +1,80 @@
+project('libdatrie', 'c', version: '1.4.0', meson_version : '>=0.53.0')
+
+fs = import('fs')
+
+sh = find_program('sh')
+compiler = meson.get_compiler('c')
+powershell = find_program('powershell', required : compiler.get_id() == 'msvc')
+cat = find_program('cat')
+deps = []
+
+if fs.exists('VERSION')
+    version = run_command(cat, files('VERSION'), check : true).stdout().strip()
+else
+    version = run_command(files('build-aux/git-version-gen'), check : true).stdout().strip()
+endif
+meson.add_dist_script(sh, '-c', 'echo "@0@" > $MESON_DIST_ROOT/VERSION'.format(version))
+meson.add_dist_script(sh, '-c', 'cd $MESON_DIST_ROOT && ./autogen.sh')
+
+conf_data = configuration_data()
+conf_data.set_quoted('VERSION', version)
+
+# Detect for iconv
+compiler_has_iconv = compiler.has_function('iconv_open')
+iconv_pkgconfig = get_option('iconv_pkgconfig')
```

Comment:
I think the flag is misnamed, it's more like use Meson's resolver to find iconv (which is required on a Mac - FindIconv is needed)

I think it should be harmless to remove the flag entirely and always use the `auto` path here?

Reply:
Does FindIconv do anything above and beyond detecting the existence of a `find_library('iconv')`?

Because having this detection not work on systems without cmake installed, is unfortunate.

Reply:
Since iconv doesn't come with pkg-config, and we only check for `iconv_open` here, it doesn't work on a Mac which needs `-liconv` to link.  FindIconv does give us the flags we need.

Reply:
Right, but `compiler.find_library('iconv')` also gives you `-liconv` without using cmake to look it up.

Reply:
Turns out [mingw-w64 does ship iconv](https://packages.msys2.org/package/mingw-w64-x86_64-libiconv). I guess I have to add back dependency support.

Reply:
That pkg-config file appears to come from the msys2 packaging, not upstream iconv. And it returns the same information that find_library() returns.

I would not bother using it since it's not an official pkg-config file.